docs/arch/Proposals/adr-2504-bulkupdate.md

```diff
@@ -0,0 +1,155 @@
+- If the search query returns no resources, a  404 "Resource Not Found" response will be returned.
+- If the search query returns multiple resources, an error indicating that the search criteria are not specific enough will be returned.
+
+### Use‑cases 
+| # | Scenario |
+|---|----------|
+| 1 | Correct a demographic error in millions of `Patient` records |
+| 2 | Add or replace security labels on many `Observation`s linked to a cohort |
+| 3 | Stamp every resource in a tenant with a provenance tag after migration |
+| 4 | Update all `ServiceRequest` records with a new category within the FHIR service without manually extracting and updating them |
+| 5 | Update `security tags` or `unit values` in bulk, based on changes announced by third-party sources |
+| 6 | Map resources referencing old `Patient` Id to new `Patient` Id |
+| 7 | `MedicationRequest` resources referencing the outdated medication need to be updated to reference the new replacement Medication |
+
+## Decision to make
+- Every patch `path` **must start with the `ResourceType` root** (e.g., `Patient.meta.tag`) to disambiguate meta‑level vs element updates. 
+- To patch the `Common properties` users should use `Resource`. (e.g., `"name": "path", "valueString": "Resource"`)
+- The bulk-update job should run to completion, with retry logic in place to handle transient, connectivity, or availability failures.
+- Transient, connectivity, or availability failures during job processing should not result in duplicate entries being added to a resource. To prevent this, updates should be processed in chunks wrapped within a transaction.
+- If a request to add an element to an array is submitted twice, duplicate entries may be created. This can be avoided by setting the `skipIfExist` query parameter flag to true. `skipIfExist` will be honored only for operation types `add` and `insert`
+- If the update request results in an invalid resource, the update will fail. If one resource becomes invalid, the update will fail for all resources of the same type.
+- Attempting to add an element to resources that already contain those elements will fail in the FHIR Patch call. The same behavior should apply to bulk-delete operations.
+- Attempting to replace an element that doesn't exist will fail in the FHIR Patch call. The same behavior should apply to bulk-delete operations.
+- If the search query returns no resources, a "Resource Not Found" response with a 404 Bad Request status will be returned. This behavior should also apply to bulk-delete operations.
+- Customers should use a `GET query to retrieve the list of resources that will be updated`.
+- To prevent log overloading, audit entries for bulk-update operations should be recorded in chunks, following the same approach used for bulk-delete operations.
+
+
+### Functional behaviour
+
+- **Invocation**  
+  `PATCH [base]/$bulk-update { | /{ResourceType}/$bulk-update }?{search-params}`  
```

Comment:
Formatting here doesn't look right. The opening { for the or clause should be in front of the /$bulk-update, not before the |.